File: mobile/types/comment.ts
export type Comment = {
    "id": number,
    "post_id": number,
    "user_id": number,
    "content": string,
    "created_at": string,
    "user": CommentAuthor
}

type CommentAuthor = {

    "id": number,
    "username": string,
    "avatar_url": string | null
}

File: backend/app/models/interactions/comment.py
from app.models.base import Base
from sqlalchemy.orm import Mapped, mapped_column, relationship
from sqlalchemy import DateTime, ForeignKey, Text
from datetime import datetime, timezone
from app.models.user import User
from app.models.post import Post

class Comment(Base):
    __tablename__ = "comments"

    id: Mapped[int] = mapped_column(primary_key=True, index=True)
    post_id: Mapped[int] = mapped_column(ForeignKey("posts.id"), index=True, nullable=False)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), index=True, nullable=False)
    content: Mapped[str] = mapped_column(Text, nullable=False)
    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), default=lambda: datetime.now(timezone.utc))
    user: Mapped["User"] = relationship(back_populates="comments")
    post: Mapped["Post"] = relationship(back_populates="comments")
